src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,231 @@
+#include "Dice.h"
+
+#include "Screen.h"
+#include "Symbols.h"
+#include "lvgl/lvgl.h"
+
+using namespace Pinetime::Applications::Screens;
+const uint8_t Dice::diceSizes[] = {2, 3, 4, 6, 8, 12, 20, 100};
+
+static void num_dice_inc_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->numDiceIncBtnEventHandler(event);
+}
+
+static void num_dice_dec_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->numDiceDecBtnEventHandler(event);
+}
+
+static void dice_size_inc_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->diceSizeIncBtnEventHandler(event);
+}
+
+static void dice_size_dec_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->diceSizeDecBtnEventHandler(event);
+}
+
+static void roll_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->rollBtnEventHandler(event);
+}
+
+void Dice::createScreen() {
+  resultCont = lv_cont_create(lv_scr_act(), nullptr);
+  lv_obj_set_size(resultCont, 240, 80);
+  lv_obj_set_pos(resultCont, 0, 0);
+  lv_cont_set_layout(resultCont, LV_LAYOUT_COLUMN_LEFT);
+  lv_obj_set_style_local_pad_all(resultCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, 10);
+
+  resultTotalLabel = lv_label_create(resultCont, nullptr);
+  lv_obj_set_size(resultTotalLabel, 220, 30);
+  lv_obj_set_style_local_text_color(resultTotalLabel, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_label_set_text(resultTotalLabel, "");
+
+  resultIndRollsLabel = lv_label_create(resultCont, nullptr);
+  lv_label_set_long_mode(resultIndRollsLabel, LV_LABEL_LONG_SROLL_CIRC);
+  lv_obj_set_size(resultIndRollsLabel, 220, 30);
+  lv_obj_set_style_local_text_color(resultIndRollsLabel, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_label_set_text(resultIndRollsLabel, "");
+
+  configCont = lv_cont_create(lv_scr_act(), nullptr);
+  lv_obj_set_size(configCont, 240, 160);
+  lv_obj_set_pos(configCont, 0, 80);
+  lv_obj_set_style_local_bg_color(configCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+
+  rollBtn = lv_btn_create(configCont, nullptr);
+  rollBtn->user_data = this;
+  lv_obj_set_event_cb(rollBtn, roll_event_handler);
+  lv_obj_set_size(rollBtn, 70, 150);
+  lv_obj_set_pos(rollBtn, 165, 5);
+  lv_obj_set_ext_click_area(rollBtn, 5, 5, 5, 5);
+  rollBtnLabel = lv_label_create(rollBtn, nullptr);
+  lv_label_set_text(rollBtnLabel, "Roll");
+
```

Comment:
You should use `lv_obj_align()` to automatically align all buttons to edges. Is there a reason to leave 5px on the edges? The buttons are already large so do we need to extend the click area? I personally wouldn't use it unless it's necessary. You could also use `lv_obj_set_style_local_value_str()` instead of labels to reduce some code.

Reply:
The 5px space outside the buttons was primarily aesthetic.  I did not have it originally and the screen looked a bit cramped with all of the buttons touching, so I added the space and extended the click area to compensate for that.  It does function without extending the click area, but I found that it felt better with it included.  As far as aligning, I found it easier to pack everything in by manually placing things -- I'm worried about getting everything to look right just using `align`.

Would `lv_obj_set_style_local_value_str()` still apply even though it's two separate lines of text?  I found it easier to break the text out into two labels so that both pieces could easily be updated separately.

Reply:
`lv_obj_set_style_local_value_str` can be used for buttons. The results label is fine.

I totally agree that buttons shouldn't be touching, but what i meant was that the buttons aren't actually touching the edge of the screen.

Reply:
Oh -- totally missed that this was for the label on the button.  I apologize.  Yeah, I can give that a shot.  And I can move the buttons out to touch the edge of the screen, I just found that the uniform padding made it a bit easier to keep everything aligned/even.

Reply:
@Riksu9000 I played around with the buttons and could not get it to look nice without the padding.  I think the math just works out better with the 5px padding.  I did add the same 5px padding to result label in https://github.com/JF002/InfiniTime/pull/565/commits/42b048d648c6f2aad3064e1760771321f0d08e19, which makes the edges of it line up with the edges of the buttons -- it looks much nicer with this.  Thoughts?

Also, I did remove the rollBtnLabel and replace it with the local value string style in the same commit.

Reply:
By using 76px buttons and 6px space between buttons everything fits perfectly. #452 and music app in #429 for reference. They also use `lv_obj_align()`.

Reply:
You could use `lv_obj_set_style_local_value_str()` to replace the increment and decrement button labels as well.

Reply:
@Riksu9000 Fixed the button size/spacing and switched to `lv_obj_set_style_local_value_str()` for all buttons in https://github.com/JF002/InfiniTime/pull/565/commits/7c0174284033c67c6e11b61d2829570715e35cb0